examples/layers/examples/signer_layer.rs

```diff
@@ -0,0 +1,66 @@
+//! Example of using the `SignerLayer` in the provider.
+
+use alloy_network::EthereumSigner;
+use alloy_node_bindings::Anvil;
+use alloy_primitives::{address, b256, U256};
+use alloy_provider::{Provider, ProviderBuilder};
+use alloy_rpc_client::RpcClient;
+use alloy_rpc_types::request::TransactionRequest;
+use alloy_signer_wallet::LocalWallet;
+use eyre::Result;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Spin up a local Anvil node.
+    // Ensure `anvil` is available in $PATH
+    let anvil = Anvil::new().try_spawn()?;
+
+    // Set up the wallets.
+    let wallet: LocalWallet = anvil.keys()[0].clone().into();
+
+    // Create a provider with the signer.
+    let http = anvil.endpoint().parse()?;
+    let provider = ProviderBuilder::new()
+        // Add the `SignerLayer` to the provider
+        .signer(EthereumSigner::from(wallet))
+        .on_client(RpcClient::new_http(http));
+
+    let tx = TransactionRequest {
+        nonce: Some(0),
+        value: Some(U256::from(100)),
+        to: address!("d8dA6BF26964aF9D7eEd9e03E53415D37aA96045").into(),
+        gas_price: Some(U256::from(20e9)),
+        gas: Some(U256::from(21000)),
+        ..Default::default()
+    };
```

Comment:
same

Reply:
Updated to use the new `TransactionBuilder`, uses a `legacy` transaction.